### PR TITLE
[IMP] sale_brand: add terms and conditions

### PR DESCRIPTION
**Pull request by switch87**
Add the possibility to add seperate sales conditions per brand

### CODE REVIEW

File: sale_brand/models/sale_order.py [hunk 1]
```diff
@@ -45,3 +46,11 @@ def default_get(self, fields_list):
             res["brand_id"] = team.brand_id.id
 
         return res
+
+    @api.model
```
Comment (sbejaoui) [inline sale_brand/models/sale_order.py]:
```suggestion
```

File: sale_brand/models/sale_order.py [hunk 2]
```diff
@@ -45,3 +46,11 @@ def default_get(self, fields_list):
...
+    def _default_note(self):
+        if self.brand_id:
+            if self.brand_id.terms_type == "html":
+                return _("Terms & Conditions: %s", self.brand_id.terms_url)
+            return self.brand_id.invoice_terms or ""
+        return super()._default_note()
```
Comment (sbejaoui) [inline sale_brand/models/sale_order.py:60]: in which base module this method exist?

File: sale_brand/tests/test_sale_order.py [hunk 3]
```diff
@@ -12,6 +12,26 @@ def setUp(self):
         self.sale.order_line.mapped("product_id").write({"invoice_policy": "order"})
         self.sale.action_confirm()
 
+    def test_terms_url(self):
+        self.sale.brand_id.write({"website": "https://odoo-community.org"})
+        self.sale.brand_id.write({"terms_type": "html"})
+        self.sale.brand_id.write(
+            {"invoice_terms": "We are not responsible for or product to work"}
+        )
+        self.sale.brand_id.write({"terms_page": "/terms"})
```
Comment (sbejaoui) [inline sale_brand/tests/test_sale_order.py]:
```suggestion
        self.sale.brand_id.write(
            {
                "website": "https://odoo-community.org",
                "terms_type": "html",
                "invoice_terms": "We are not responsible for or product to work",
                "terms_page": "/terms",
            }
        )
```

File: sale_brand/tests/test_sale_order.py [hunk 4]
```diff
@@ -12,6 +12,26 @@ def setUp(self):
...
+        self.assertEqual(
+            self.sale.brand_id.terms_url, "https://odoo-community.org/terms"
+        )
+        self.assertEqual(
+            self.sale._default_note(),
```
Comment (sbejaoui) [inline sale_brand/tests/test_sale_order.py]: call the onchange instead

**Review by sbejaoui — CHANGES_REQUESTED** [review]

File: sale_brand/models/sale_order.py (hunk 2 above)
Comment (switch87) [reply sale_brand/models/sale_order.py:60]: the method exists in the sale module

Comment (sbejaoui) [reply sale_brand/models/sale_order.py:60]: sorry I can't find it, can you give me a link,

Comment (switch87) [reply sale_brand/models/sale_order.py:60]: https://github.com/odoo/odoo/blob/de707dd3751c4d0fa21f7961057daffb6f437d93/addons/sale/models/sale_order.py#L111

Comment (sbejaoui) [reply sale_brand/models/sale_order.py:60]: thanks

**Review by sbejaoui — APPROVED** [review]
LGTM, thanks

File: sale_brand/models/sale_order.py [hunk 5]
```diff
@@ -45,3 +46,10 @@ def default_get(self, fields_list):
             res["brand_id"] = team.brand_id.id
 
         return res
+
+    def _default_note(self):
+        if self.brand_id:
+            if self.brand_id.terms_type == "html":
```
Comment (sbejaoui) [inline sale_brand/models/sale_order.py]:
```suggestion
        use_invoice_terms = self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
        if use_invoice_terms and self.brand_id and self.brand_id.terms_type == "html":
```

**Review by sbejaoui — CHANGES_REQUESTED** [review]
we should check that `use_invoice_terms` is True like base method

**Comment by sbejaoui** [comment]
please check pre-commit

**Comment by switch87** [comment]
> please check pre-commit

It's a long regex that I re-used from Odoo, I don't know how and if I can split that over multiple lines?

Edit: Sorry, didn't see there where new errors involving me using ' al the time instead of " ;) will solve that

**Comment by switch87** [comment]
> > please check pre-commit
> 
> It's a long regex that I re-used from Odoo, I don't know how and if I can split that over multiple lines?
> 
> Edit: Sorry, didn't see there where new errors involving me using ' al the time instead of " ;) will solve that

now its only the regex. If you know how to fix it let me know. Googling it only gives me how to split lines using a regex.
@sbejaoui

**Comment by sbejaoui** [comment]
> > > please check pre-commit
> > 
> > 
> > It's a long regex that I re-used from Odoo, I don't know how and if I can split that over multiple lines?
> > Edit: Sorry, didn't see there where new errors involving me using ' al the time instead of " ;) will solve that
> 
> now its only the regex. If you know how to fix it let me know. Googling it only gives me how to split lines using a regex. @sbejaoui

I see, it's easier to ignore it.

please merge this PR in yours: https://github.com/SnakeByteDevelopment/brand/pull/1

**Review by sbejaoui — APPROVED** [review]

**Review by sbejaoui — APPROVED** [review]

**Comment by switch87** [comment]
@sbejaoui Hi Bejaoui, what's the next step for this pr? Something I need to do?

**Comment by sbejaoui** [comment]
> @sbejaoui Hi Bejaoui, what's the next step for this pr? Something I need to do?

LGTM, I waited for a second review,

**Review by sbejaoui — APPROVED** [review]

**Pull request merged**